src/System.Composition/tests/InheritanceTests.cs

```diff
@@ -57,6 +58,51 @@ public void ExportsAtTheClassLevelAreAppliedIgnoringBaseExports()
             Assert.True(cc.TryGetExport(out derivedExport));
         }
 
+        [Fact]
+        public void GetExport_WithoutWhereClause_ExportSuccessful()
+        {
+            CompositionContext container = CreateContainer(
+                typeof(IMefCollection),
+                typeof(IMefCollection<,>),
+                typeof(BaseClass),
+                typeof(DerivedClass),
+                typeof(MefCollection<,>)
+                );
+            IMefCollection<DerivedClass, BaseClass> actualValue;
+            Assert.True(container.TryGetExport(out actualValue));
+            Assert.NotNull(actualValue);
+            Assert.IsType<MefCollection<DerivedClass, BaseClass>>(actualValue);
+        }
+
+        [Fact]
+        [ActiveIssue(23607)]
```

Comment:
Adding these two new tests revealed the bug exists in System.Composition.

I am adding ActiveIssue on this test becauase Assert.True() statement on line 89 is failing.

Reply:
Is your plan here to fix the issue or add tests that prove that the behavior is indeed not working?

Reply:
My plan was to set it for future and work on the fix after 2.1 release

Reply:
As long as we can confirm it isn't a regression then I'm fine doing it post 2.1.

Reply:
Just checked now. It is a regression.

Reply:
@weshaggard @ViktorHofer this is the PR that caused this regression: https://github.com/dotnet/corefx/pull/10219 and please refer to the last comment:

> Thinking back now, I believe the reason we didn't tackle this was the difficulties inherent in checking more complex constraints like:
>
> `class AResponder<T,U> : IResponder<T,U> where T: U {}`
I don't think failing in these cases is a reason not to proceed with this improvement, but good to keep in mind if/when future bug reports pop up. (If there is a straightforward way to lift this limitation it would be great to cover all cases, but I'm not aware of any current API that can check these rigorously.)

We would have to remove the changed in PR dotnet/corefx#10219 to fix the failing test here.